### PR TITLE
[WIP] Coverity fixes

### PR DESCRIPTION
**Pull request by Therzok**
* [ ] Guard against exceptions in GettextCatalog.Get(Plural)String, so we don't end up with lots of false positives.
* [ ] More to come.

### CODE REVIEW

File: main/src/addins/NUnit/Gui/NUnitOptionsPanel.cs [hunk 1]
```diff
@@ -166,7 +166,7 @@ void OnRemoveCategory (object sender, EventArgs args)
 		void OnCategoryEdited (object sender, EditedArgs args)
 		{
 			TreeIter iter;
-			if (!store.GetIter (out iter, new TreePath (args.Path)))
+			if (!store.GetIterFromString (out iter, args.Path))
```
Comment (Therzok) [inline main/src/addins/NUnit/Gui/NUnitOptionsPanel.cs]: This was a GLib Opaque which doesn't seem to have any unmanaged memory reclaimed on finalize.

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/RootWorkspace.cs [hunk 2]
```diff
@@ -1572,7 +1572,7 @@ public void NotifyChanges ()
 				FileService.NotifyFilesChanged (modified);
 		}
 		
-		void IDisposable.Dispose ()
+		public void Dispose ()
```
Comment (slluis) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/RootWorkspace.cs]: Let's keep this private. That's not something we want to be visible in the public api.

**Comment by slluis** [comment]
Other than the comment, LGTM

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/RootWorkspace.cs (hunk 2 above)
Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/RootWorkspace.cs]: But why is it marked as Disposable then? Dispose isn't called, so why not just move it on the Finalizer?

Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/RootWorkspace.cs]: Ah, nevermind, the finalizer might not work correctly on Mono (it's not precise gc-ing). why not just drop the Dispose method them? Makes no sense to hide it and have the type as IDisposable.

Comment (slluis) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/RootWorkspace.cs]: RootWorkspace implements IWorkspaceObject, which has IDisposable as base interface, so dispose has to be implemented.

Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/RootWorkspace.cs]: But we're in the `FileStatusTracker` class, not RootWorkspace. How is it related?

Comment (slluis) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/RootWorkspace.cs]: Ah, right, forget it then

**Pull request merged**